src/compiler/scanner.rs

```diff
@@ -33,6 +43,24 @@ impl Scanner {
                     self.set_pos(self.pos() + 1);
                     return self.set_token(SyntaxKind::AsteriskToken);
                 }
+                CharacterCodes::_0
+                | CharacterCodes::_1
+                | CharacterCodes::_2
+                | CharacterCodes::_3
+                | CharacterCodes::_4
+                | CharacterCodes::_5
+                | CharacterCodes::_6
+                | CharacterCodes::_7
+                | CharacterCodes::_8
+                | CharacterCodes::_9 => {
+                    let ScanNumberReturn {
+                        type_: token,
+                        value: token_value,
+                    } = self.scan_number();
```

Comment:
The more natural way to represent these one-off compound return types in Rust would presumably be to return a tuple but I'm trying to mimic the Typescript code as closely as I can, so creating these one-off struct types for structured return values